### PR TITLE
HDDS-10578. Clean up Hadoop dependencies

### PR DESCRIPTION
**Pull request by myskov**
## What changes were proposed in this pull request?
HDDS-10578. Clean up Hadoop dependencies

There are a lot of Hadoop dependencies, and a significant portion of them are now obsolete. 
I removed unused dependencies.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-10578

## How was this patch tested?

Existing tests

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -341,44 +341,11 @@ xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xs
...
-        <groupId>org.apache.hadoop</groupId>
-        <artifactId>hadoop-client-modules</artifactId>
-        <version>${hadoop.version}</version>
-        <type>pom</type>
-      </dependency>
       <dependency>
         <groupId>org.apache.hadoop</groupId>
         <artifactId>hadoop-client-api</artifactId>
         <version>${hadoop.version}</version>
       </dependency>
```
Comment (adoroszlai) [inline pom.xml:348]: I think this one is also unused.

File: pom.xml [hunk 2]
```diff
@@ -416,238 +378,17 @@ xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xs
...
-        <version>${hadoop.version}</version>
-        <type>test-jar</type>
-      </dependency>
-
       <dependency>
         <groupId>org.apache.hadoop</groupId>
         <artifactId>hadoop-mapreduce-client-jobclient</artifactId>
         <version>${hadoop.version}</version>
         <type>test-jar</type>
       </dependency>
```
Comment (adoroszlai) [inline pom.xml:391]: This one, too.

**Review by adoroszlai — COMMENTED** [review]
Thanks @myskov for the cleanup.  I think some more hadoop dependencies can be removed, but otherwise LGTM.

**Review by kerneltime — APPROVED** [review]

File: pom.xml (hunk 2 above)
Comment (adoroszlai) [reply pom.xml:386]: Root POM defines two similar dependencies:

https://github.com/apache/ozone/blob/0a405d70f1e0b78176a8580c491593d1aa2a2436/pom.xml#L376-L386

but only one of them is used (the one without `<type>test-jar`):

https://github.com/apache/ozone/blob/0a405d70f1e0b78176a8580c491593d1aa2a2436/hadoop-ozone/integration-test/pom.xml#L207-L211

**Pull request merged**